tsdb/engine/tsm1/engine.go

```diff
@@ -475,6 +477,10 @@ func (e *Engine) WithLogger(log zap.Logger) {
 
 // LoadMetadataIndex loads the shard metadata into memory.
 func (e *Engine) LoadMetadataIndex(shardID uint64, index tsdb.Index) error {
+	if index.Type() != inmem.IndexName {
```

Comment:
Instead of doing this check, what do you think about calling a func on the `Index` itself.  The `inmem` would do this and `tsi` would be a `hoop`.  I think adding `index.Type() == foo` could lead to complicated logic in the code over time.